src/wallet/wallet.h

```diff
@@ -1112,10 +1112,10 @@ class WalletRescanReserver
 };
 
 //! Add wallet name to persistent configuration so it will be loaded on startup.
-bool AddWalletSetting(interfaces::Chain& chain, const std::string& wallet_name);
+bool AddWalletSetting(WalletContext& context, const std::string& wallet_name);
```

Comment:
Would be good to guard against lock contention (+ for `RemoveWalletSetting`):

<details>
<summary>git diff on 135eb1d5b1</summary>

```diff
diff --git a/src/wallet/wallet.h b/src/wallet/wallet.h
index ad2e2817c4..fd61b39c47 100644
--- a/src/wallet/wallet.h
+++ b/src/wallet/wallet.h
@@ -19,6 +19,7 @@
 #include <script/script.h>
 #include <support/allocators/secure.h>
 #include <sync.h>
+#include <threadsafety.h>
 #include <tinyformat.h>
 #include <uint256.h>
 #include <util/fs.h>
@@ -28,6 +29,7 @@
 #include <util/time.h>
 #include <util/ui_change_type.h>
 #include <wallet/crypter.h>
+#include <wallet/context.h>
 #include <wallet/db.h>
 #include <wallet/scriptpubkeyman.h>
 #include <wallet/transaction.h>
@@ -1112,10 +1114,10 @@ public:
 };
 
 //! Add wallet name to persistent configuration so it will be loaded on startup.
-bool AddWalletSetting(WalletContext& context, const std::string& wallet_name);
+bool AddWalletSetting(WalletContext& context, const std::string& wallet_name) EXCLUSIVE_LOCKS_REQUIRED(!context.wallets_mutex);
 
 //! Remove wallet name from persistent configuration so it will not be loaded on startup.
-bool RemoveWalletSetting(WalletContext& context, const std::string& wallet_name);
+bool RemoveWalletSetting(WalletContext& context, const std::string& wallet_name) EXCLUSIVE_LOCKS_REQUIRED(!context.wallets_mutex);
 
 struct MigrationResult {
     std::string wallet_name;

```
</details>

Reply:
Done.